packages/compass-indexes/src/modules/indexes.ts

```diff
@@ -0,0 +1,209 @@
+import IndexModel from 'mongodb-index-model';
+import type { Document } from 'mongodb';
+import { localAppRegistryEmit } from '@mongodb-js/mongodb-redux-common/app-registry';
+import type { ThunkAction, ThunkDispatch } from 'redux-thunk';
+import _debug from 'debug';
+
+import type { RootState } from './index';
+import { handleError } from './error';
+import type { HandleErrorAction, IndexesError } from './error';
+import { ActionTypes as RefreshActionTypes } from './is-refreshing';
+import type { RefreshFinishedAction } from './is-refreshing';
+
+const debug = _debug('mongodb-compass:modules:indexes');
+
+type SortField = keyof Pick<
+  IndexDefinition,
+  'name' | 'type' | 'size' | 'usageCount' | 'properties'
+>;
+export type SortColumn = keyof typeof sortColumnToProps;
+export type SortDirection = 'asc' | 'desc';
+const sortColumnToProps: Record<string, SortField> = {
+  'Name and Definition': 'name',
+  Type: 'type',
+  Size: 'size',
+  Usage: 'usageCount',
+  Properties: 'properties',
+};
```

Comment:
```suggestion
const sortColumnToProps = {
  'Name and Definition': 'name',
  Type: 'type',
  Size: 'size',
  Usage: 'usageCount',
  Properties: 'properties',
} as const;
```

to just directly get the exact type?